tokio/src/task/waker.rs

```diff
@@ -18,16 +19,7 @@ where
     T: Future,
     S: Schedule,
 {
-    let ptr = meta as *const _ as *const ();
-
-    let vtable = &RawWakerVTable::new(
-        clone_waker::<T, S>,
-        wake_unreachable,
-        wake_by_local_ref::<T, S>,
-        noop,
-    );
-
-    let waker = unsafe { Waker::from_raw(RawWaker::new(ptr, vtable)) };
+    let waker = unsafe { ManuallyDrop::new(Waker::from_raw(raw_waker::<T, S>(meta))) };
```

Comment:
Can you maybe add a link to https://github.com/rust-lang/rust/issues/66281 in a comment that explains why we are reusing the vtable.
And a short blurb like:

> This reuses the `wake()` and `drop()` functions from the owned `waker()`.
> However this does not matter, since the methods can never be called on the waker reference.

Reply:
I added the comment :+1: